dev/benchmarking/ddl/1_02_table_public_gpu.sql

```diff
@@ -0,0 +1,35 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+*/
+
+CREATE EXTENSION IF NOT EXISTS "citext";  -- type for case-insensitive text
+
+-- GPU
+CREATE TABLE IF NOT EXISTS public.gpu
+(
+  gpu_id SERIAL PRIMARY KEY
+  , gpu_model_name citext UNIQUE NOT NULL DEFAULT ''
```

Comment:
If I understand the objective of this PR correctly then the aim is to characterize the benchmark machine with those parameters that may affect the benchmark result. When looking at the GPU parameters then it is not always obvious if the core count or RAM size will actually influence the benchmark results. And there may be other relevant parameters that may have affect to the benchmark results. For instance, when ECC is enabled then the effective RAM size will be smaller than reported in specs. Since the relevance of all device parameters is not clear, I would suggest recording the full output of `nvidia-smi -q` so that analysis tools can pick up the relevant parameters as needed. To me, choosing just the core count and RAM size sounds a little arbitrary..

As for "GPU model" question, perhaps "GPU Part Number" would characterize the GPU device in a more unique way.. but keep "GPU Product Name" (="GPU model") as the part number is not descriptive. 
Just my 2c.

Reply:
Thank you both! Very, very helpful! I changed the "gpu" table to have the columns "gpu_part_number", "gpu_part_name", and "gpu_nvidia_smi" which is maybe a bad name but it's the only one I could think of just now.

Reply:
I just realized that the gpu table here is NVIDIA specific but there is also AMD GPU and other devices that may have different tools to inquire the device parameters. In addition, the `nvidia-smi` program is available only for Linux and Windows systems, but not for Mac OSX.
On Mac OSX, I have used `kextstat | grep -i cuda` to get the NVIDIA cards information. IIRC, this did not include information about "gpu_part_number". However, there is also https://github.com/phvu/cuda-smi that outputs the minimal information about device. It uses CUDA RT to retrieve the device information.
For this issue, perhaps it would make sense to implement a similar tool that outputs device parameters for Linux, Windows, and Mac OSX in the same format?
In addition, CUDA is not the only library that can be used for managing NVIDIA GPU devices, there is also OpenCL and AMD ROCm. So, I would be careful using term "gpu" when one actually thinks of "cuda".

Suggestion: replace "gpu_nvidia_smi" with a more generic name, say, "gpu_information". Its content may be the output of a any program that is gives the relevant information. For instance, the program could be:
- `nvidia-smi` - when using CUDA on Linux and Windows
- `cuda-smi` or `kextstat | grep -i cuda` - when using CUDA on Mac OSX
- ??? - when using Open CL with NVIDIA, AMD GPU, or other devices
- ??? - when using AMD ROCm with AMD GPU devices
- etc

Reply:
In CUDA, `deviceQuery` in CUDA samples gives a pretty good summary of the device, it looks even better than `nvidia-smi -q` output.

Reply:
Everyone in this discussion has helped me a lot. I am grateful for your time. The entries in the `gpu` table are now `gpu_information` (unique default ''), `gpu_part_number` (default ''), and `gpu_product_name` (default ''). There was discussion [in the listserv](https://lists.apache.org/thread.html/0e41b8dd84672bbbc5644ecdfe8eb2bdf09d6e868a8c754f5f8603a9@%3Cdev.arrow.apache.org%3E) about making `gpu_information` the only value in the table. That's fine. As a compromise I made it the unique index and people who don't want to use other columns can just make them empty strings.